src/printer.js

```diff
@@ -2330,19 +2337,46 @@ function nodeStr(node, options) {
     shouldUseSingleQuote = true;
   }
 
-  const result = jsesc(str, {
-    quotes: shouldUseSingleQuote ? "single" : "double",
-    wrap: true
-  });
+  const raw = options.parser === 'flow' ? node.raw : node.extra.raw;
```

Comment:
In many other places we do property checks instead of hardcoding a type of parser. What do you think about doing `node.extra ? node.extra.raw : node.raw` instead, or something like that?